modules/files.nix

```diff
@@ -27,11 +54,23 @@ in
       type = fileType "<envar>HOME</envar>" homeDirectory;
     };
 
+    home-file-defs = mkOption {
+      type = with types; listOf attrs;
+      internal = true;
+      description = "All home file definitions, each having kind <literal>fileType</literal>";
+    };
+
     home-files = mkOption {
       type = types.package;
       internal = true;
       description = "Package to contain all home files";
     };
+
+    homeManager.fileType = mkOption {
```

Comment:
With https://github.com/rycee/home-manager/commit/f8aaba6704818ef6af55d00909d5eb6130cbd3be perhaps it is possible to add something like `lib.fileType = fileType;` or even `lib.types.file = fileType;` to the `config` section and use `type = config.fileType;` in the xdg module?

Reply:
Yes, great idea!